FrontEnd/src/App.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import './App.css';
+
+import * as Msal from 'msal'
+
+
+class App extends React.Component {
+  componentDidMount() {
+    if (msalApp.getAccount()) {
+      showInfo();
+    }
+  }
+
+  render() {
+    return (
+      <div className="App">
+        <div>
+          <button onClick={login}>Logg inn</button>
+          <button onClick={logout}>Logg ut</button>
+        </div>
+        <div>
+          <p id="tokenName"></p>
+          <p id="tokenId"></p>
+        </div>
+      </div>
+    );
+  }
+}
+
+var msalConfig = {
+  auth: {
+    clientId: "e90cbb61-896e-4ec7-aa37-23511700e1ed",
+    authority: "https://login.microsoftonline.com/3aa4a235-b6e2-48d5-9195-7fcf05b459b0"
+  },
+  cache: {
+    cacheLocation: "localStorage",
+    storeAuthStateInCookie: true
+  }
+};
+
+var msalApp = new Msal.UserAgentApplication(msalConfig);
+
+function login() {
+  const loginRequest = {
+    scopes: ["user.read"]/*, "user.write"*/
+  };
+
+  msalApp.loginPopup(loginRequest).then(function (loginResponse) {
+    //login success
+    console.log(loginResponse)
+    showInfo();
+  })
+    .catch(function (error) {
+      console.log(error);
+    });
+}
+
+function logout() {
+  msalApp.logout();
+}
+
+function showInfo() {
+  document.getElementById("tokenName").innerText = msalApp.getAccount().name;
```

Comment:
This is a super no-no for React. React uses av Virtual DOM so it only updates the Browser DOM when the new version if different than the old one and then only the parts that changes. If you updates the Browser DOM underneath React you will be fighting it every step of the way. And super wired bugs will emerge.

You most likely wants want reacts call a Stateful Component.